webproject/src/main/java/com/google/sps/dataManagers/RoomManager.java

```diff
@@ -0,0 +1,81 @@
+package com.google.sps.dataManagers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.ServletException;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseReference;
+import com.google.sps.data.CategoryProto.Category;
+import com.google.sps.data.RoomProto.Room;
+import com.google.sps.util.FirebaseUtil;
+
+@Singleton
+public class RoomManager {
+    private FirebaseUtil firebaseUtil;
+
+    @Inject
+    public RoomManager(FirebaseUtil firebaseUtil) {
+        this.firebaseUtil = firebaseUtil;
+    }
+
+    public Queue<Room> getAllRooms() throws ServletException {
+        DatabaseReference ref = firebaseUtil.getRoomsReference();
+        List<DataSnapshot> dataSnapshots = firebaseUtil.getAllSnapshotsFromReference(ref);
+
+        Queue<Room> sortedRooms = new PriorityQueue<>(dataSnapshots.size(), (aRoom, bRoom) -> {
+            double aRoomAveragePerPersonValue =
+                    (aRoom.getMinPrice() - aRoom.getOrdersValue() + aRoom.getDeliveryFee()) / aRoom.getUsersCount();
+            double bRoomAveragePerPersonValue =
+                    (bRoom.getMinPrice()  - bRoom.getOrdersValue() + bRoom.getOrdersValue()) / bRoom.getUsersCount();
+            return  Double.compare(aRoomAveragePerPersonValue, bRoomAveragePerPersonValue);
+        });
+
+        for (DataSnapshot dataSnapshot : dataSnapshots) {
+            String id = dataSnapshot.getKey();
```

Comment:
do you think this will be reusable? maybe we can put it into Room class e.g loadFromDataSnapshot()

Reply:
okay!